### PR TITLE
[CRIMAPP-265] Add council tax page

### PR DESCRIPTION
**Pull request by arthurashman**
## Description of change
Add council tax page

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-265

## Screenshots of changes (if applicable)
image

## How to manually test the feature

1. rails db:migrate
2. Start an application
3. Go to steps/outgoings/does_client_pay_council_tax
4. Check that submitting the form without selecting an option displays error 'Select yes if your client pays Council Tax where they usually live'
5. Check that submitting the form with Yes selected but no amount number input displays error 'Enter how much your client pays yearly'
6. Check that submitting with Yes selected and an amount input submits ok

### CODE REVIEW

File: app/services/decisions/income_decision_tree.rb [hunk 1]
```diff
@@ -24,10 +24,11 @@ def destination
       when :delete_dependant
         edit_dependants
       when :dependants_finished
-        edit('/home', action: :index)
+        # TODO: link to next step when we have it
+        edit(:manage_without_income)
       when :manage_without_income
         # TODO: link to next step when we have it
-        show('/home', action: :index)
```
Comment (arthurashman) [inline app/services/decisions/income_decision_tree.rb:30]: Just to keep user testers in flow whilst this is all in flux

File: config/locales/en/errors.yml [hunk 2]
```diff
@@ -323,6 +323,17 @@ en:
...
+          attributes:
+            housing_payment_type:
+              inclusion: Select which payment your client pays where they usually live
+        steps/outgoings/council_tax_form:
+          attributes:
+            pays_council_tax:
+              inclusion: Select yes if your client pays Council Tax where they usually live
+            council_tax_amount:
+              blank: Enter how much your client pays yearly
+              not_a_number: How much they pay yearly must be a number, like 50
```
Comment (arthurashman) [inline config/locales/en/errors.yml:336]: I don't know how to actually hit this as a user, especially with the field in the html set as a `govuk_number_field`, _and_ `inputmode: 'numeric'` 
 It does show if I remove `presence: true ` from the `council_tax_amount` validation in the form
 
I also can't hit it in the spec when passing in a string. 

Don't think this is a huge issue as the form UI doesn't allow for non-numeric input

File: app/services/decisions/income_decision_tree.rb [hunk 3]
```diff
@@ -24,7 +24,8 @@ def destination
       when :delete_dependant
         edit_dependants
       when :dependants_finished
-        edit('/home', action: :index)
```
Comment (arthurashman) [inline app/services/decisions/income_decision_tree.rb:27]: Just to link everything together

File: app/views/steps/income/lost_job_in_custody/edit.html.erb [hunk 4]
```diff
@@ -3,26 +3,28 @@
```
Comment (arthurashman) [inline app/views/steps/income/lost_job_in_custody/edit.html.erb:3]: Sorry about this one, just fixing indenting!

File: app/forms/steps/outgoings/council_tax_form.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    class CouncilTaxForm < Steps::BaseFormObject
+      include Steps::HasOneAssociation
+      has_one_association :outgoings
+
+      attribute :pays_council_tax, :value_object, source: YesNoAnswer
+      attribute :council_tax_amount, :integer
+
+      validates_inclusion_of :pays_council_tax, in: :choices
+
+      validates :council_tax_amount,
```
Comment (willmcb) [inline app/forms/steps/outgoings/council_tax_form.rb:12]: One thing you can do here is say 'Yes' to paying council tax and enter zero and the form submits, thinking that may be good to have a validation that the value entered needs to be > 0? 🤔

File: config/locales/en/errors.yml (hunk 2 above)
Comment (willmcb) [reply config/locales/en/errors.yml:336]: I don't think you can get to it via the UI

File: app/forms/steps/outgoings/council_tax_form.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+module Steps
+  module Outgoings
+    class CouncilTaxForm < Steps::BaseFormObject
+      include Steps::HasOneAssociation
+      has_one_association :outgoings
+
+      attribute :pays_council_tax, :value_object, source: YesNoAnswer
+      attribute :council_tax_amount, :integer
```
Comment (willmcb) [inline app/forms/steps/outgoings/council_tax_form.rb:8]: One thing here is that by having it as an integer we lose pence amounts. Not sure how much of an issue this is but you can input pounds and pence and submit and then when you come back the pence amount after the decimal point is lost.

**Review by willmcb — CHANGES_REQUESTED** [review]
Would like to discuss the comments here if thats okay - they are only minor

File: app/forms/steps/outgoings/council_tax_form.rb (hunk 5 above)
Comment (arthurashman) [reply app/forms/steps/outgoings/council_tax_form.rb:14]: 👍 Now negative or 0 doesn't pass validation and shows error message

File: app/forms/steps/outgoings/council_tax_form.rb (hunk 6 above)
Comment (arthurashman) [reply app/forms/steps/outgoings/council_tax_form.rb]: 👍 Changed to string here, and in db table. Will convert to pence integer in the serialization before sending to datastore, and back again on rehydration

Comment (arthurashman) [reply app/forms/steps/outgoings/council_tax_form.rb:8]: Storing now as council_tax_amount in pence

**Review by willmcb — APPROVED** [review]
Have rerun a regression and tested the decimal issue and it all works great :)

**Pull request merged**